### PR TITLE
#15, #16, #14 - Refactors InstaGramSearchBy* and adds SearchByLocation

### PR DESCRIPTION
**Pull request by oleiman**
Signed-off-by: Oren Leiman <oleiman@neutral.io>

### CODE REVIEW

**Comment by hansmosh** [comment]
@oleiman It looks to me like this PR can be closed as all these commits are already in #19 
https://github.com/nio-blocks/instagram/pull/19

Can you please verify that for me and Close this PR if it is true.

File: instagram_search_by.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+    def configure(self, context):
+        super().configure(context)
+        lb = self._unix_time(datetime.utcnow() - self.lookback)
+        self._freshest = [lb] * self._n_queries
+        # Convert queries from usernames to ids.
+        self.queries = [i for i in [self._process_query(q)
+                        for q in self.queries] if i]
+        # reset n in case some usernames did not convert to ids.
+        self._n_queries = len(self.queries) or 1
```
Comment (hansmosh) [inline instagram_search_by.py]: I think this "or 1" used to exist in the rest polling block and was copied over from that, but I don't think it really makes sense. If their are no queries then it should be 0.

File: instagram_search_by.py [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+            resp = self._make_request(resource_url)
+            if resp is None:
+                # if the response is still bad, give up.
+                return
+
+        resources = resp.json().get('data', [])
+        return self._extract_resource_id(resources, query)
+
+    def _construct_resource_url(self, query):
+        return self.RESOURCE_URL_FORMAT.format(
```
Comment (hansmosh) [inline instagram_search_by.py]: It seems odd that the base block doesn't have a RESOURCE_URL_FORMAT even though it is used in the block. I get that blocks that use this should set it but the base block should probably have a placeholder one.

File: instagram_search_by_radius.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    def _process_query(self, query):
+        """ Convert LocationRadius objects into lat/lng api parameters.
+
+        """
+        params = "lat={0}&lng={1}&distance={2}"
+        return params.format(query.latitude, query.longitude, query.radius)
+
+    @property
+    def current_query(self):
+        return self.queries[self._idx]
```
Comment (hansmosh) [inline instagram_search_by_radius.py]: Good call here.

**Comment by hansmosh** [comment]
I like the approach. I just have those few comments to be reviewed.

**Comment by hansmosh** [comment]
Since none of the comments are crucial, I could merge this, then merge #19 and file issues for those few things.

**Comment by oleiman** [comment]
nah, i'll go through 'em now.

File: instagram_search_by.py (hunk 1 above)
Comment (oleiman) [inline instagram_search_by.py]: i think you're right.

File: instagram_search_by.py (hunk 2 above)
Comment (oleiman) [inline instagram_search_by.py]: Ya. Unintentional omissions. The latter is easily fixed, the former has some grey area. I'll have to mull it over.

**Pull request merged**